File: java/difido-reports/src/main/java/il/co/topq/difido/model/execution/Node.java
package il.co.topq.difido.model.execution;



import il.co.topq.difido.model.Enums.Status;

import org.codehaus.jackson.annotate.JsonIgnore;
import org.codehaus.jackson.annotate.JsonSubTypes;
import org.codehaus.jackson.annotate.JsonSubTypes.Type;
import org.codehaus.jackson.annotate.JsonTypeInfo;

@JsonTypeInfo(
	    use = JsonTypeInfo.Id.NAME,
	    include = JsonTypeInfo.As.PROPERTY,
	    property = "type")
	@JsonSubTypes({
	    @Type(value = ScenarioNode.class, name = "scenario"),
	    @Type(value = TestNode.class, name = "test"),
	    @Type(value = MachineNode.class, name = "machine") })

public abstract class Node {

	private String name;

	private Status status = Status.success;

	@JsonIgnore
	private NodeWithChildren parent;

	public Node() {

	}

	public Node(String name) {
		this.name = name;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public Status getStatus() {
		return status;
	}

	public void setStatus(Status status) {
		if (getStatus().ordinal() < status.ordinal()) {
			this.status = status;

		}
		if (getParent() != null){
			getParent().setStatus(status);
		}
	}

	public NodeWithChildren getParent() {
		return parent;
	}

	public void setParent(NodeWithChildren parent) {
		this.parent = parent;
	}

}

File: binders/difido-nunit3/difido-common/Model/Execution/Node.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using System.Web.Script.Serialization;

namespace Difido.Model.Execution
{
    public abstract class Node
    {
        private string statusValue = "success";
        
        public string name { get; set; }
        
        public string status { get {return statusValue;}
            
            set 
            {
                if (statusValue == null){
                    statusValue = value;
                    if (parent != null)
                    {
                        parent.status = value;
                    }
                    return;
                }
                Status currentStatus = (Status)Enum.Parse(typeof(Status), statusValue);
                Status newStatus = (Status)Enum.Parse(typeof(Status), value);
                if (currentStatus < newStatus)
                {
                    statusValue = value;
                    if (parent != null)
                    {
                        parent.status = value;
                    }
                    

                }
            } 
        
        }
        public string type { get; set; }

        [ScriptIgnoreAttribute]
        public NodeWithChildren parent { get; set; }


        public Node()
        {
        }

        public Node(string name)
        {
            this.name = name;
        }


    }
}
